src/org/zaproxy/zap/extension/frontendscanner/FrontEndScannerAPI.java

```diff
@@ -44,6 +66,44 @@ public String getPrefix() {
 
     @Override
     public String handleCallBack(HttpMessage msg) throws ApiException {
+        try {
+            String query = msg.getRequestHeader().getURI().getQuery();
+
+            if (query == null) {
+                throw new ApiException (ApiException.Type.MISSING_PARAMETER, "action");
+            }
+
+            Map<String, String> parameters = Stream.of(query.split("&"))
+              .map(paramAndValue -> paramAndValue.split("="))
+              .collect(Collectors.toMap(
+                  e -> e[0],
+                  e -> e[1]
```

Comment:
This will lead to an exception if the parameter has no value. I'd suggest using one of the core classes/methods to obtain the parameters already parsed, e.g.:
```Java
Map<String, String> parameters = new StandardParameterParser().getParams(msg, HtmlParameter.Type.url);
```
(`new StandardParameterParser()` could be made an instance variable.)

Reply:
Great! Was looking for something like that but did not find it...